### PR TITLE
commander + vitest for cli

### PR DESCRIPTION
**Pull request by haochuan**
This one took me a while and sorry for the long PR, I feel it's a little hard to separate those stuff into smaller PRs.

This PR is for:
- migrate `oclif` to `commander`
- migrate `mocha` to `vitest`
- clean up the pain of `git reset` (resetting all of your current uncommitted files during test)
- add a quick doc for the cli structure

### CODE REVIEW

**Comment by lynchbomb** [comment]
Yeah this is looking good. Likely once this PR has landed we can set the CLI up for scale by extending the `Command` class with a base RehearsalCLI class. Essentially we want the UX across the CLI to match especially as it pertains to flags and positional args. 

eg. we dont want the `upgrade` cmd to have `--dir` and `migrate` to have `--cwd` essentially both flags are synonyms

**Comment by egorio** [comment]
@haochuan currently cli tests interact with git (git reset) which is painful, please try to get rid of this. :) You can follow *.input / *.output way to test as we do in other packages.

**Comment by haochuan** [comment]
@egorio yeah I got a lot of troubles when running tests with uncommitted files, will cleanup the gut stuff

**Review by lynchbomb — APPROVED** [review]

**Review by egorio — APPROVED** [review]

File: packages/cli/src/commands/migrate.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env node
+
+import { Command } from 'commander';
+const program = new Command();
+
+program.parse(process.argv);
+
+console.log('migrate');
```
[the remark this answers is not in the document]
Comment (haochuan) [reply packages/cli/src/commands/migrate.ts:8]: @lynchbomb I'll clean this up in my next `migrate` command PR

**Pull request merged**